test/swap.test.ts

```diff
@@ -1,984 +1,488 @@
-        initialFields: {
-          invariant: invariant.contractId,
-          token0: token0.contractInstance.contractId,
-          token1: token1.contractInstance.contractId,
-          fee: fee,
-          tickSpacing: tickSpacing,
-          xToY: true,
-          amount: swapAmount,
-          byAmountIn: true,
-          sqrtPriceLimit: slippage
-        },
-        tokens: [
-          { id: tokenX.contractInstance.contractId, amount: swapAmount },
-          { id: tokenY.contractInstance.contractId, amount: swapAmount }
-        ]
-      })
-
-      const swapperTokenXBalanceAfter = await balanceOf(
-        tokenX.contractInstance.contractId,
-        swapper.address
-      )
-      const swapperTokenYBalanceAfter = await balanceOf(
-        tokenY.contractInstance.contractId,
-        swapper.address
-      )
-      const invariantTokenXBalanceAfter = await balanceOf(
-        tokenX.contractInstance.contractId,
-        invariant.address
-      )
-      const invariantTokenYBalanceAfter = await balanceOf(
-        tokenY.contractInstance.contractId,
-        invariant.address
+      await initPositionWithLiquidity(
+        invariant,
+        positionOwner,
+        tokenX,
+        positionAmount,
+        tokenY,
+        positionAmount,
+        fee,
+        tickSpacing,
+        lowerTickIndex - 20n,
+        middleTickIndex,
+        liquidity,
+        2n,
+        slippageLimit,
+        slippageLimit
       )
+    }
 
-      expect(swapperTokenXBalanceAfter).toBe(0n)
-      expect(swapperTokenYBalanceAfter).toBe(1990n)
-      expect(invariantTokenXBalanceAfter).toBe(1500n)
-      expect(invariantTokenYBalanceAfter).toBe(1509n)
-
-      const poolAfter = decodePool(
-        (
-          await invariant.methods.getPool({
-            args: {
-              token0: token0.contractInstance.contractId,
-              token1: token1.contractInstance.contractId,
-              fee: fee,
-              tickSpacing: tickSpacing
-            }
-          })
-        ).returns
-      )
+    // swap + check results
+    {
+      const swapper = await getSigner(ONE_ALPH * 1000n, 0)
+      await withdrawTokens(swapper, [tokenY, swapAmount])
 
-      const positionLiquidity = 1000000n * 10n ** 5n
-      expect(poolAfter.liquidity - positionLiquidity).toBe(poolBefore.liquidity)
-      expect(poolAfter.currentTickIndex).toBe(-20n)
-      expect(poolAfter.sqrtPrice).toBe(999254456240199142700995n)
-      expect(poolAfter.feeGrowthGlobalX).toBe(40000000000000000000000n)
-      expect(poolAfter.feeGrowthGlobalY).toBe(0n)
-      expect(poolAfter.feeProtocolTokenX).toBe(2n)
-      expect(poolAfter.feeProtocolTokenY).toBe(0n)
+      const dexBalance = {
+        tokenX: await balanceOf(tokenX.contractId, invariant.address),
+        tokenY: await balanceOf(tokenY.contractId, invariant.address)
+      }
+      expect(dexBalance).toStrictEqual({ tokenX: 500n, tokenY: 2499n })
+
+      expectError(
+        InvariantError.LimitReached,
```

Comment:
Durning a swap there can be two limits reached, tick and sqrtPrice. Maybe it could be a good idea to make the error more precise?

Reply:
I was thinking that too, there's another code for `PriceLimitReached`, so I will prefix the current one with `Tick`.
I assume this is a leftover from moving error codes around, the original code might have been akin to "TickmapError.LimitReached'